### PR TITLE
PR to resolve remaining warnings left in eslint

### PR DESCRIPTION
**Pull request by shellyear**
Note:  There are no actual eslint errors remained, but the other ones described [here](https://github.com/kbss-cvut/record-manager-ui/issues/113)

### CODE REVIEW

**Comment by shellyear** [comment]
terminal after running **npm run validate**

I suggest to solve the chunk size warning in a separate branch. Created [issue](https://github.com/kbss-cvut/record-manager-ui/issues/117) for that 

Screenshot 2024-03-08 at 11 34 55

File: vite.config.js [hunk 1]
```diff
@@ -18,6 +18,14 @@ export default defineConfig({
   build: {
     sourcemap: true,
     emptyOutDir: true,
+    rollupOptions: {
+      onwarn: (warning, defaultHandler) => {
+        if (warning.code === "INVALID_ANNOTATION" && warning.message.includes("*#__PURE__*")) {
```
Comment (blcham) [inline vite.config.js]: Let's create an issue in SForms, referencing this commit and other related links.

File: vite.config.js [hunk 2]
```diff
@@ -26,6 +26,7 @@ export default defineConfig({
         defaultHandler(warning);
       },
     },
+    cssMinify: false,
```
Comment (blcham) [inline vite.config.js:36]: What's the issue here? we say temporarily, but we do not describe the issue here, nor some relevant discussion where we can assume that a fix will appear.

File: vite.config.js [hunk 3]
```diff
@@ -14,6 +15,7 @@ export default defineConfig({
     eslintPlugin({
       cache: false, // disable eslint cache to avoid conflicts
     }),
+    nodePolyfills(),
```
Comment (blcham) [inline vite.config.js:18]: Are we able to describe what is the issue? We could reference not only issue but also what causes it

It seems that it is again related to lib `parse-link-headers`. We can discuss alternatives to this library. Maybe some fixed fork?

File: vite.config.js [hunk 4]
```diff
@@ -22,7 +22,12 @@ export default defineConfig({
     emptyOutDir: true,
     rollupOptions: {
       onwarn: (warning, defaultHandler) => {
-        if (warning.code === "INVALID_ANNOTATION" && warning.message.includes("*#__PURE__*")) {
+        // TODO: remove isFromKbssCvutPackageWarning and solve the root cause, see https://github.com/kbss-cvut/record-manager-ui/issues/113
+        const isFromKbssCvutPackageWarning =
```
Comment (blcham) [inline vite.config.js:26]: ?

**Review by blcham — CHANGES_REQUESTED** [review]
Nicely separated in commits. My comments mostly suggest adding some information to the found issues -- so they are trackable.

File: vite.config.js [hunk 5]
```diff
@@ -12,12 +13,27 @@ export default defineConfig({
...
   ],
   build: {
     sourcemap: true,
     emptyOutDir: true,
+    rollupOptions: {
+      onwarn: (warning, defaultHandler) => {
+        // TODO: remove isFromKbssCvutPackageWarning and solve the root cause, see https://github.com/kbss-cvut/record-manager-ui/issues/113
+        const isFromKbssCvutPackageWarning =
+          warning.code === "EVAL" && warning.message.includes("node_modules/store/plugins/lib/json2.js");
+        const isRollupPureAnnotationWarning =
```
Comment (blcham) [inline vite.config.js:28]:
```suggestion
        // TODO: Rollup Pure Annotation warning should be resolved by https://github.com/kbss-cvut/s-forms/issues/282 
        const isRollupPureAnnotationWarning =
```

File: vite.config.js (hunk 1 above)
Comment (blcham) [reply vite.config.js]: Will be resolved by https://github.com/kbss-cvut/s-forms/issues/282.

File: vite.config.js (hunk 3 above)
Comment (blcham) [reply vite.config.js:18]: again we need to reference here issue which we can watch to solve the issue

File: vite.config.js [hunk 6]
```diff
@@ -12,12 +13,27 @@ export default defineConfig({
...
+      cache: false,
     }),
+    nodePolyfills(),
   ],
   build: {
     sourcemap: true,
     emptyOutDir: true,
+    rollupOptions: {
+      onwarn: (warning, defaultHandler) => {
+        // TODO: remove isFromKbssCvutPackageWarning and solve the root cause, see https://github.com/kbss-cvut/record-manager-ui/issues/113
```
Comment (blcham) [inline vite.config.js:25]: This is not very informative. I do not understand why you are filtering this. Please write the reason here if it is  ok to filter or try to find solution without filtering the warning.

File: vite.config.js (hunk 5 above)
Comment (shellyear) [reply vite.config.js:26]: Left a comment [here](https://github.com/kbss-cvut/record-manager-ui/issues/113#issuecomment-1995671719), but we can continue in this thread

Comment (blcham) [reply vite.config.js:49]: I believe the comment above is not related to the line that I annotated here:
` const isRollupPureAnnotationWarning =`

I am committing my suggesting here.

File: index.html [hunk 7]
```diff
@@ -4,7 +4,6 @@
     <meta charset="UTF-8" />
     <link rel="icon" href="favicon.ico" />
     <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=0" />
-    <script src="config.js"></script>
```
Comment (blcham) [inline index.html:7]: image

I did change as indicated above to fix build warning ... so rollup `htmlPlugin` might not be needed. Is there any catch to that solution?
I tested the solution in `deploy/internal-auth` at it works for me.

@shellyear Could you please check if it is viable solution?

File: vite.config.js [hunk 8]
```diff
@@ -12,14 +12,34 @@ export default defineConfig({
...
+      },
+    },
+    cssMinify: false,
   },
   define: {
     "process.env": process.env, // workaround for parse-link-header library that depends on 2 vars defined in `process.env`, see https://github.com/thlorenz/parse-link-header/issues/31
   },
+  resolve: {
+    alias: {
+      querystring: "querystring-es3",
```
Comment (blcham) [inline vite.config.js:41]:
```suggestion
      querystring: "querystring-es3", // workaround for parse-link-header library that replaces nodejs builtin module with the module adapted for browser
```

File: vite.config.js [hunk 9]
```diff
@@ -12,14 +12,34 @@ export default defineConfig({
...
+      url: "url-parse",
```
Comment (blcham) [inline vite.config.js:42]:
```suggestion
      url: "url-parse", // workaround for parse-link-header library that replaces nodejs builtin module with the module adapted for browser
```

File: vite.config.js [hunk 10]
```diff
@@ -12,14 +12,34 @@ export default defineConfig({
...
   ],
   build: {
     sourcemap: true,
     emptyOutDir: true,
+    rollupOptions: {
+      onwarn: (warning, defaultHandler) => {
+        // TODO: remove isFromKbssCvutPackageWarning and solve the root cause, see https://github.com/kbss-cvut/record-manager-ui/issues/113
+        const isFromKbssCvutPackageWarning =
+          warning.code === "EVAL" && warning.message.includes("node_modules/store/plugins/lib/json2.js");
+        const isRollupPureAnnotationWarning =
```
Comment (blcham) [inline vite.config.js:26]:
```suggestion
        // TODO: Rollup Pure Annotation warning should be resolved by https://github.com/kbss-cvut/s-forms/issues/282 
        const isRollupPureAnnotationWarning =
```

File: vite.config.js [hunk 11]
```diff
@@ -12,14 +12,34 @@ export default defineConfig({
...
+          warning.code === "INVALID_ANNOTATION" && warning.message.includes("*#__PURE__*");
+        if (isFromKbssCvutPackageWarning || isRollupPureAnnotationWarning) {
+          return;
+        }
+        defaultHandler(warning);
+      },
+    },
+    cssMinify: false,
```
Comment (blcham) [inline vite.config.js:34]:
```suggestion
    cssMinify: false, // TODO: workaround for CSS syntax error from SForms library that should be resolved by https://github.com/kbss-cvut/s-forms/issues/283
```

File: vite.config.js [hunk 12]
```diff
@@ -12,14 +12,34 @@ export default defineConfig({
...
-      cache: false, // disable eslint cache to avoid conflicts
+      cache: false,
     }),
   ],
   build: {
     sourcemap: true,
     emptyOutDir: true,
+    rollupOptions: {
+      onwarn: (warning, defaultHandler) => {
+        // TODO: remove isFromKbssCvutPackageWarning and solve the root cause, see https://github.com/kbss-cvut/record-manager-ui/issues/113
```
Comment (blcham) [inline vite.config.js:23]:
```suggestion
        // TODO: workaround for SForms issue that uses eval method that is discouraged. It should be solved in https://github.com/kbss-cvut/s-forms/issues/283
```

File: vite.config.js [hunk 13]
```diff
@@ -12,14 +12,34 @@ export default defineConfig({
...
+        // TODO: Rollup Pure Annotation warning should be resolved by https://github.com/kbss-cvut/s-forms/issues/282
+        const isRollupPureAnnotationWarning =
+          warning.code === "INVALID_ANNOTATION" && warning.message.includes("*#__PURE__*");
+        if (isFromKbssCvutPackageWarning || isRollupPureAnnotationWarning) {
+          return;
+        }
+        defaultHandler(warning);
+      },
+    },
+    cssMinify: false,
```
Comment (blcham) [inline vite.config.js:34]:
```suggestion
    cssMinify: false, // TODO: workaround for CSS syntax error from SForms library that should be resolved by https://github.com/kbss-cvut/s-forms/issues/283
```

File: vite.config.js [hunk 14]
```diff
@@ -12,14 +12,34 @@ export default defineConfig({
...
-      cache: false, // disable eslint cache to avoid conflicts
+      cache: false,
     }),
   ],
   build: {
     sourcemap: true,
     emptyOutDir: true,
+    rollupOptions: {
+      onwarn: (warning, defaultHandler) => {
+        // TODO: workaround for dangerous use of eval method that should be solved in SForms library issue https://github.com/kbss-cvut/s-forms/issues/283        const isFromKbssCvutPackageWarning =
```
Comment (blcham) [inline vite.config.js:23]:
```suggestion
        // TODO: workaround for dangerous use of eval method that should be solved in SForms library issue https://github.com/kbss-cvut/s-forms/issues/283
        const isFromKbssCvutPackageWarning =
```

**Pull request merged**